sdk/tables/Azure.Data.Tables/MigrationGuide.md

```diff
@@ -0,0 +1,309 @@
+
+To try and improve the development experience across Azure services, including Tables, a set of uniform
+[design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent
+experience with established API patterns for all services. A set of
+[.NET-specific guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html) was also introduced to ensure that .NET clients have a natural and idiomatic
+feel that mirrors that of the .NET base class libraries. Further details are available in the guidelines for those interested.
+
+The new Azure Tables library `Azure.Data.Tables` provides the ability to share in some of the cross-service improvements made to the Azure development experience.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespace root for the modern Azure client libraries for .NET have changed. Each will follow the pattern `Azure.[Area].[Services]` where the
+legacy clients followed the pattern `Microsoft.Azure.[Service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using
+the modern or legacy clients.
+
+In the case of Tables, the modern client library is named `Azure.Data.Tables` and was released beginning with version 12. The legacy client libraries have packages
+and namespaces that begin with `Microsoft.Azure.CosmosDB` or `Microsoft.Azure.Storage` and a version of 2.x.x or below and 9.x.x or below respectively.
+
+### Constructing the clients
+
+Previously in `Microsoft.Azure.Comsmos.Table`, you would create a `CloudStorageAccount` which can be used to get an instance of the `CloudTableClient` in order to
+perform service level operations.
+
+```C#
+// Create the CloudStorageAccount using StorageCredentials.
+CloudStorageAccount storageAccount = new CloudStorageAccount(
+    new StorageCredentials(accountName, storageAccountKey),
+    storageUri);
+ 
+ // Get a reference to the cloud table client.
+ CloudTableClient tableClient = storageAccount.CreateCloudTableClient(new TableClientConfiguration());
+```
+
+Now, in `Azure.Data.Tables`, we only need a `TableServiceClient` for service level operations.
+
+```C# Snippet:TablesSample1CreateClient
+// Construct a new <see cref="TableServiceClient" /> using a <see cref="TableSharedKeyCredential" />.
+
+var serviceClient = new TableServiceClient(
+    new Uri(storageUri),
+    new TableSharedKeyCredential(accountName, storageAccountKey));
+```
+
+### Creating a table
+
+Previously in `Microsoft.Azure.Comsmos.Table`, we'd use a `CloudTable` instance to create a table, which is returned from the `CloudTableClient` method on
+`CloudTableClient`.
+
+```C#
+// Create a table client and create the table if it doesn't already exist.
+string tableName = "OfficeSupplies1p1";
+CloudTable table = tableClient.GetTableReference(tableName);
+cloudTable.CreateIfNotExists()
```

Comment:
`cloudTable` is used on L73 but `table` is defined on L72.     Is that a mistake?

Reply:
yes - good catch - this was a rename that went wrong :)